src/util.py

```diff
@@ -338,36 +389,27 @@ def as_version_sort_key(v):
     return int(a[0]) * 1000000 + int(a[1]) * 1000 + int(a[2])
 
 
-def get_latest_as_version(as_major_version):
+def get_latest_as_version(as_major_version, as_channel):
     """Find the last A-S version on Maven for the given major version"""
 
-    # Find the latest release in the multi-arch .aar
-
-    # TODO What is the right package to check here? full-megazord metadata seems broken.
-    r = requests.get(f"{MAVEN}/org/mozilla/appservices/nimbus/maven-metadata.xml")
-    r.raise_for_status()
-    metadata = xmltodict.parse(r.text)
-
-    versions = []
-    for version in metadata["metadata"]["versioning"]["versions"]["version"]:
-        if version.startswith(f"{as_major_version}."):
-            versions.append(version)
-
-    if len(versions) == 0:
-        raise Exception(f"Could not find any A-S {as_major_version} releases")
-
-    latest = max(versions, key=as_version_sort_key)
-
-    # TODO Make sure this release has been uploaded for all architectures.
-
-    return latest
+    if as_channel == "nightly":
+        r = requests.get(
+            taskcluster_indexed_artifact_url(
+                "project.application-services.v2.nightly.latest",
+                "public/build/nightly.json",
+            )
+        )
+        r.raise_for_status()
+        return r.json()["version"]
+    else:
+        raise NotImplementedError("Only the AS nightly channel is currently supported")
```

Comment:
Ah, relbot doesn't "ride the train". Any change we make and release has to be compatible with nightly, beta, and release at once. I'm sorry about this. The good news is: this logic will go away once [bug android-to-gecko-repos](https://bugzilla.mozilla.org/show_bug.cgi?id=android-to-gecko-repos) is done